tracing-subscriber/src/filter/env/mod.rs

```diff
@@ -28,7 +28,44 @@ use tracing_core::{
 
 /// A `Layer` which filters spans and events based on a set of filter
 /// directives.
-// TODO(eliza): document filter directive syntax?
+///
+/// # Directives
+///
+/// A filter consists of one or more directives. Directives match [`Span`]s and [`Event`]s
+/// and specify a maximum verbosity [`level`] to enable for those that match. The directive
+/// syntax is similar to the one presented in `env_logger`. The syntax consists
+/// of four parts `target[span{field=value}]=level`.
+///
+/// - `target` matches on the target that the event comes from, generally this will be the
+/// module path. Examples, `h2`, `tokio::net`, etc. For more info checkout the [`Metadata`] docs.
+/// - `span` matches on the span name that you want to filter on. If this is supplied with a `target`
+/// it will match on all filter spans within that `target.
+/// - `field` matches the fields within spans. Field names can also be supplied without a `value`
+/// and will match on any `Span` that contains that field name.
+/// Example, `[span{field=\"value\"}]=debug`, `[{field}]=trace`, etc.
+/// - `value` matches the _output_ of the span's value. If a value is a numeric literal or a bool,
+// it will match that value only. Otherwise, it's a regex that matches the `std::fmt::Debug` output
+/// from the value. Examples, `1`, `\"some_string\"`, etc.
+/// - `level` sets a maximum verbosity level accepted by this directive
+///
+/// The portion of the synatx that is included within the square brackets is `tracing` specific.
+/// All portions of the syntax are omissable. If a `value` is provided a `field`
+/// must be specified. If just a `level` is provided it will enable all `Span`s and `Event`s.
```

Comment:
```suggestion
/// must be specified. If just a level is provided, it will set the maximum level for all `Span`s and `Event`s that are not enabled by other filters.
```